tools/KoreBuild.Console/Commands/SubCommandBase.cs

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Tools.Internal;
+
+namespace KoreBuild.Console.Commands
+{
+    internal class SubCommandBase : CommandBase
+    {
+        private const string _defaultToolsSource = "https://aspnetcore.blob.core.windows.net/buildtools";
+        private const string _dotnetFolderName = ".dotnet";
+
+        private CommandOption _repoPathOption;
+        private CommandOption _dotNetHomeOption;
+        private CommandOption _toolsSourceOption;
+        private CommandOption _verbose;
+
+        private string _koreBuildDir;
+
+        public string KoreBuildDir => _koreBuildDir;
+        public string ConfigDirectory => Path.Combine(KoreBuildDir, "config");
+        public string RepoPath => _repoPathOption.HasValue() ? _repoPathOption.Value() : Directory.GetCurrentDirectory();
+        public string DotNetHome => GetDotNetHome();
+        public string ToolsSource => _toolsSourceOption.HasValue() ? _toolsSourceOption.Value() : _defaultToolsSource;
+        public string SDKVersion => GetDotnetSDKVersion();
+
+        public IReporter Reporter => new ConsoleReporter(PhysicalConsole.Singleton, _verbose != null, false);
+
+        public override void Configure(CommandLineApplication application)
+        {
+            _koreBuildDir = FindKoreBuildDirectory();
+
+            base.Configure(application);
+
+            _verbose = application.Option("-v|--verbose", "Show verbose output", CommandOptionType.NoValue);
+            _toolsSourceOption = application.Option("--toolsSource", "The source to draw tools from.", CommandOptionType.SingleValue);
```

Comment:
Nit: command-line typically use kebab-case, so `--tools-source` and `--dotnet-home`. See build.sh for example.

https://github.com/aspnet/BuildTools/blob/a8447f5472f2e0b96c85322edccc10d6e84a01f9/scripts/bootstrapper/build.sh#L31-L37